Max/src/main/java/springbootApp/java/models/InstrumentoFactory.java

```diff
@@ -0,0 +1,20 @@
+package springbootApp.java.models;
+
+import org.springframework.stereotype.Component;
+
+public class InstrumentoFactory {
+    public static InstrumentoFinanciero nuevoInstrumento(String nombre, double precio, Tipo tipo) {
+        InstrumentoFinanciero i;
+        switch (tipo) {
+            case Tipo.ACCION:
+                i = new Accion(nombre, precio);
+                break;
+            case Tipo.BONO:
+                i = new Bono(nombre, precio);
+                break;
+            default:
+                throw new IllegalArgumentException("Error. Tipo invalido");
+        }
+        return i;
+    }
```

Comment:
Yo se que parece una pavada, pero en lugar de declararla con el identificador "i" queda mas legible que sea instrumentoFinanciero:

```suggestion
public class InstrumentoFactory {
    public static InstrumentoFinanciero nuevoInstrumento(String nombre, double precio, Tipo tipo) {
        InstrumentoFinanciero instrumentoFinanciero;
        switch (tipo) {
            case Tipo.ACCION:
                instrumentoFinanciero = new Accion(nombre, precio);
                break;
            case Tipo.BONO:
                instrumentoFinanciero = new Bono(nombre, precio);
                break;
            default:
                throw new IllegalArgumentException("Error. Tipo invalido");
        }
        return instrumentoFinanciero;
    }
```